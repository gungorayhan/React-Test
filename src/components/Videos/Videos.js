import React from 'react'

const Videos = ({videos =[]}) => {
    if(videos.length <= 0){
        return <h2>Soryy, no videos found</h2>
    }

    return (
        <div>
            <h3>Videos</h3>
            {videos.map((video,index)=>(
                <h1 key={index}>{video}</h1>
            ))}
        </div>
    )
}

export default Videos