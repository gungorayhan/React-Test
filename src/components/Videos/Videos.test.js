import React from "react"
import Videos from "./Videos"
import renderer from "react-test-renderer"

test("Videos empty renders ",()=>{
    const tree= renderer.create(<Videos/>).toJSON();

    expect(tree).toMatchSnapshot();
})

test("Videos 1 item renders",()=>{
    const tree= renderer.create(<Videos videos={["Udemy"]}/>).toJSON();

    expect(tree).toMatchSnapshot();
})