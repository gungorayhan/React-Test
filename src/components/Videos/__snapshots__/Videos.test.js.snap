// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos 1 item renders 1`] = `
<div>
  <h3>
    Videos
  </h3>
  <h1>
    Udemy
  </h1>
</div>
`;

exports[`Videos empty renders  1`] = `
<h2>
  Soryy, no videos found
</h2>
`;
